src/perform_step/extrapolation_perform_step.jl

```diff
@@ -190,16 +190,16 @@ function prediction!(integrator,cache::RichardsonEulerConstantCache,temp_dt)
 end
 
 function prediction!(integrator,cache::RichardsonEulerCache,temp_dt)
```

Comment:
This is definitely not thread-safe. Every thread is writing to the same cache.

Reply:
Every call of `prediction!` writes to `T[n,1]` only and for each thread, value of `n` will be different due to different `temp_dt`. Also an instance of `ktilde` is created to act as a dummy `k` in function. I am also not using `u` from cache to write intermediate value of `u` in starting. Though this will definetly increase allocation in this function.
Can there be any other problems in this function which I am not considering?

Reply:
ooh, I see. I read the `@.` on the wrong side. Yes, it would be worthwhile to create a cache for that temporary, one for each thread. That's an optimization that can be handled in a future PR, but is a very necessary one.

Reply:
I'll create a new PR with optimization for threading.